rplidar.go

```diff
@@ -113,30 +134,28 @@ func (rp *Rplidar) NextPointCloud(ctx context.Context) (pointcloud.PointCloud, e
 
 // start requests that the rplidar starts up and starts spinning.
 func (rp *Rplidar) start() {
-	rp.mu.Lock()
-	defer rp.mu.Unlock()
-
 	rp.started = true
-	rp.logger.Debug("starting motor")
-	rp.device.driver.StartMotor()
+	if rp.model != "S1" {
+		rp.logger.Debug("starting motor")
+		rp.device.driver.StartMotor()
+	}
 	rp.device.driver.StartScan(false, true)
 	rp.nodes = gen.New_measurementNodeHqArray(rp.nodeSize)
 }
 
 // stop request that the rplidar stops spinning.
 func (rp *Rplidar) stop() {
-	rp.mu.Lock()
```

Comment:
[note to self] Stop gets called in Close. 2 weeks ago Kim implemented a change that Close takes the **same** mutex. So right now the goroutines are leaking because stop isn't able to acquire the mutex.

Reply:
Could it be useful to leave a comment that if these functions will be used, you must lock the mutex first?

Reply:
Functions have been placed in line so this shouldn't be an issue